libgadget/sfr_eff.c

```diff
@@ -447,298 +192,68 @@ cooling_direct(int i) {
     }
 }
 
-#if defined(SFR)
-
-
-/* returns 0 if the particle is actively forming stars */
-static int get_sfr_condition(int i) {
-    int flag = 1;
-/* no sfr !*/
+/* returns 1 if the particle is on the effective equation of state,
+ * cooling via the relaxation equation and maybe forming stars.
+ * 0 if the particle does not form stars, instead cooling normally.*/
+static int
+sfreff_on_eeqos(int i)
+{
+    int flag = 0;
+    /* no sfr: normal cooling*/
     if(!All.StarformationOn) {
-        return flag;
+        return 0;
     }
-    if(SPHP(i).Density * All.cf.a3inv >= All.PhysDensThresh)
-        flag = 0;
-
-    if(SPHP(i).Density < All.OverDensThresh)
-        flag = 1;
-
-    /* massless particles never form stars! */
+    /* Check that starformation has not left a massless particle.*/
     if(P[i].Mass == 0) {
-        endrun(-1, "Encoutered zero mass particle during sfr ;"
-                  " We haven't implemented tracer particles and this shall not happen\n");
+        endrun(-1, "Particle %d in SFR has mass=%g. Does not happen as no tracer particles.\n", i, P[i].Mass);
     }
 
-    if(SPHP(i).DelayTime > 0)
-        flag = 1;		/* only normal cooling for particles in the wind */
-
-    if(All.QuickLymanAlphaProbability > 0) {
-        double dloga = get_dloga_for_bin(P[i].TimeBin);
-        double unew = DMAX(All.MinEgySpec,
-                (SPHP(i).Entropy + SPHP(i).DtEntropy * dloga) /
-                GAMMA_MINUS1 * pow(SPHP(i).EOMDensity * All.cf.a3inv, GAMMA_MINUS1));
-
-        const double u_to_temp_fac = (4 / (8 - 5 * (1 - HYDROGEN_MASSFRAC))) * PROTONMASS / BOLTZMANN * GAMMA_MINUS1
-        * All.UnitEnergy_in_cgs / All.UnitMass_in_g;
-
-        double temp = u_to_temp_fac * unew;
+    if(SPHP(i).Density * All.cf.a3inv >= All.PhysDensThresh)
+        flag = 1;
 
-        if(SPHP(i).Density > All.OverDensThresh && temp < 1.0e5)
-            flag = 0;
-        else
-            flag = 1;
-    }
+    if(SPHP(i).Density < All.OverDensThresh)
+        flag = 0;
 
+#ifdef SFR
+    if(SPHP(i).DelayTime > 0)
+        flag = 0;		/* only normal cooling for particles in the wind */
+#endif
     return flag;
 }
 
-/*These functions are for the wind models*/
-static int
-sfr_wind_weight_haswork(int target, TreeWalk * tw)
-{
-    if(P[target].Type == 4) {
-        if(P[target].IsNewParticle && !P[target].DensityIterationDone) {
-             return 1;
-        }
-    }
-    return 0;
-}
-
-static int
-sfr_wind_feedback_haswork(int target, TreeWalk * tw)
+static int make_particle_star(int i, double mass_of_star)
```

Comment:
Please describe the meaning of retflag.

Reply:
ok, will do. If it helps retflag disappears in the later (feature) commits